core/src/robot/AbstractTransmission.cc

```diff
@@ -177,40 +178,29 @@ namespace jiminy
             }
         }
 
-        jointIndex_t jointModelIdxTmp;
-        for (unsigned int i = 0; i < jointNames_.size(); i++)
-        {
-            if (returnCode == hresult_t::SUCCESS)
-            {
-                // jointModelIndices_ is empty at this point, therefore inject a 0 and later get the right index
-                jointModelIndices_.push_back(jointModelIdxTmp);
-                returnCode = ::jiminy::getJointModelIdx(robot->pncModel_, jointNames_[i], jointModelIndices_[i]);
-            }
-        }
-
-        joint_t jointType;
-        for (unsigned int i = 0; i < jointNames_.size(); i++)
-        {
-            if (returnCode == hresult_t::SUCCESS)
-            {
-                // jointTypes_ is empty at this point, therefore inject a 0 and later get the right type
-                jointTypes_.push_back(jointType);
-                returnCode = getJointTypeFromIdx(robot->pncModel_, jointModelIndices_[i], jointTypes_[i]);
-            }
-        }
-
-        for (unsigned int i = 0; i < jointNames_.size(); i++)
-        {
-            if (returnCode == hresult_t::SUCCESS)
-            {
-                // Transmissions are only supported for linear and rotary joints
-                if (jointTypes_[i] != joint_t::LINEAR && jointTypes_[i] != joint_t::ROTARY && jointTypes_[i] != joint_t::ROTARY_UNBOUNDED)
-                {
-                    PRINT_ERROR("A transmission can only be associated with a 1-dof linear or rotary joint.");
-                    returnCode = hresult_t::ERROR_BAD_INPUT;
-                }
-            }
-        }
+        std::transform(jointNames_.begin(), jointNames_.end(),
```

Comment:
the conclusion was to not use transform here at the end x)

Reply:
hahahahahaha damn it... I misunderstood your sentence:

>I would say do you use it if the capture is not empty.

Now I guess you wanted to say do not use it :-D

Reply:
ahah yes indeed. Fixed the typo for posterity.